src/main/java/io/github/christophermanahan/carnitas/Router.java

```diff
@@ -38,32 +43,42 @@ public HTTPResponse handle(HTTPRequest request) {
           .orElseGet(handler(request));
     }
 
-    private Predicate<Route> matches(HTTPRequest request) {
+    private Predicate<Route> matches(Request request) {
         return route -> route.equals(new Route(request.method(), request.uri()));
     }
 
-    private Supplier<HTTPResponse> handler(HTTPRequest request) {
-        List<HTTPRequest.Method> allowed = allowed(request);
+    private Supplier<Response> handler(Request request) {
+        List<String> allowed = allowed(request);
         return allowed.isEmpty() ? notFound() : not(allowed);
     }
 
-    private List<HTTPRequest.Method> allowed(HTTPRequest request) {
+    private List<String> allowed(Request request) {
         return map.keySet().stream()
           .filter(route -> route.uri().equals(request.uri()))
           .map(Route::method)
+          .map(Enum::toString)
+          .sorted()
           .collect(Collectors.toList());
     }
 
-    private Supplier<HTTPResponse> not(List<HTTPRequest.Method> allowed) {
+    private Supplier<Response> not(List<String> allowed) {
         return new ResponseBuilder()
-          .setStatus(HTTPResponse.Status.METHOD_NOT_ALLOWED)
-          .addHeader(Headers.contentLength(0))
-          .addHeader(Headers.allow(allowed));
+          .set(Response.Status.METHOD_NOT_ALLOWED)
+          .add(Headers.CONTENT_LENGTH + 0)
+          .add(Headers.ALLOW + String.join(" ", allowed));
     }
 
-    private Supplier<HTTPResponse> notFound() {
+    private Supplier<Response> notFound() {
         return new ResponseBuilder()
-          .setStatus(HTTPResponse.Status.NOT_FOUND)
-          .addHeader(Headers.contentLength(0));
+          .set(Response.Status.NOT_FOUND)
+          .add((Headers.CONTENT_LENGTH + 0));
+    }
+
+    private Function<Request, Response> options() {
+        return (Request request) -> new ResponseBuilder()
+          .set(Response.Status.OK)
+          .add(Headers.ALLOW + String.join(" ", allowed(request)))
+          .add(Headers.CONTENT_LENGTH + 0)
+          .get();
```

Comment:
The cohesion of this class is decreasing I think. There are more and more methods that are unrelated (in a way). We might look at extracting some of these methods at some point in the future.

For instance, `options`, `notFound` and some sort of `notAllowed` handler could all be extracted to their own classes. There is also some sort of route building in there too.

Reply:
I plan on extracting the route building when I work on the redirect story. that may also be a good opportunity to extract the classes you have mentioned.